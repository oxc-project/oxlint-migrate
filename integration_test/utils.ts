import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
