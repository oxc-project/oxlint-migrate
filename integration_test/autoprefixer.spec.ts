import { expect, test } from 'vitest';
// @ts-ignore
import autoprefixer_test from './projects/autoprefixer.eslint.config.mjs';
import { getSnapshotResult, getSnapShotMergeResult } from './utils.js';

test('autoprefixer', async () => {
  const result = await getSnapshotResult(autoprefixer_test);
  expect(result).toMatchSnapshot('autoprefixer');
});

test('autoprefixer merge', async () => {
  const result = await getSnapShotMergeResult(autoprefixer_test, {
    categories: {
      correctness: 'error',
      perf: 'error',
    },
  });
  expect(result).toMatchSnapshot('autoprefixer--merge');
});
