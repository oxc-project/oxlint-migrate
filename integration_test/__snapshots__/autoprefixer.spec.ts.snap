// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`autoprefixer > autoprefixer 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
      "commonjs": true,
      "es2022": true,
    },
    "globals": {
      "AbortController": "readonly",
      "AbortSignal": "readonly",
      "Blob": "readonly",
      "BroadcastChannel": "readonly",
      "Buffer": "readonly",
      "ByteLengthQueuingStrategy": "readonly",
      "CompressionStream": "readonly",
      "CountQueuingStrategy": "readonly",
      "Crypto": "readonly",
      "CryptoKey": "readonly",
      "CustomEvent": "readonly",
      "DOMException": "readonly",
      "DecompressionStream": "readonly",
      "Event": "readonly",
      "EventTarget": "readonly",
      "File": "readonly",
      "FormData": "readonly",
      "Headers": "readonly",
      "MessageChannel": "readonly",
      "MessageEvent": "readonly",
      "MessagePort": "readonly",
      "PerformanceEntry": "readonly",
      "PerformanceMark": "readonly",
      "PerformanceMeasure": "readonly",
      "PerformanceObserver": "readonly",
      "PerformanceObserverEntryList": "readonly",
      "PerformanceResourceTiming": "readonly",
      "ReadableByteStreamController": "readonly",
      "ReadableStream": "readonly",
      "ReadableStreamBYOBReader": "readonly",
      "ReadableStreamBYOBRequest": "readonly",
      "ReadableStreamDefaultController": "readonly",
      "ReadableStreamDefaultReader": "readonly",
      "Request": "readonly",
      "Response": "readonly",
      "SubtleCrypto": "readonly",
      "TextDecoder": "readonly",
      "TextDecoderStream": "readonly",
      "TextEncoder": "readonly",
      "TextEncoderStream": "readonly",
      "TransformStream": "readonly",
      "TransformStreamDefaultController": "readonly",
      "URL": "readonly",
      "URLSearchParams": "readonly",
      "WebAssembly": "readonly",
      "WritableStream": "readonly",
      "WritableStreamDefaultController": "readonly",
      "WritableStreamDefaultWriter": "readonly",
      "__dirname": "readonly",
      "__filename": "readonly",
      "atob": "readonly",
      "btoa": "readonly",
      "clearImmediate": "readonly",
      "clearInterval": "readonly",
      "clearTimeout": "readonly",
      "console": "readonly",
      "crypto": "readonly",
      "document": "readonly",
      "fetch": "readonly",
      "navigator": "readonly",
      "performance": "readonly",
      "process": "readonly",
      "queueMicrotask": "readonly",
      "setImmediate": "readonly",
      "setInterval": "readonly",
      "setTimeout": "readonly",
      "structuredClone": "readonly",
      "window": "readonly",
    },
    "ignorePatterns": [
      "coverage",
    ],
    "overrides": [
      {
        "files": [
          "**/*.test.{js,jsx}",
          "**/*.test.{ts,tsx}",
        ],
        "rules": {
          "no-unused-expressions": "off",
        },
      },
    ],
    "plugins": [
      "import",
      "promise",
    ],
    "rules": {
      "array-callback-return": [
        "error",
        {
          "allowImplicit": false,
          "checkForEach": false,
        },
      ],
      "constructor-super": "error",
      "default-case-last": "error",
      "eqeqeq": [
        "error",
        "always",
        {
          "null": "ignore",
        },
      ],
      "for-direction": "error",
      "getter-return": "error",
      "import/export": "error",
      "import/first": "error",
      "import/no-duplicates": "error",
      "import/no-named-default": "error",
      "import/no-webpack-loader-syntax": "error",
      "new-cap": [
        "error",
        {
          "capIsNew": false,
          "newIsCap": true,
          "properties": true,
        },
      ],
      "no-array-constructor": "error",
      "no-async-promise-executor": "error",
      "no-caller": "error",
      "no-case-declarations": "error",
      "no-class-assign": "error",
      "no-compare-neg-zero": "error",
      "no-cond-assign": "error",
      "no-console": "off",
      "no-const-assign": "error",
      "no-constant-condition": [
        "error",
        {
          "checkLoops": false,
        },
      ],
      "no-control-regex": "error",
      "no-debugger": "error",
      "no-delete-var": "error",
      "no-dupe-class-members": "error",
      "no-dupe-else-if": "error",
      "no-dupe-keys": "error",
      "no-duplicate-case": "error",
      "no-empty": [
        "error",
        {
          "allowEmptyCatch": true,
        },
      ],
      "no-empty-character-class": "error",
      "no-empty-pattern": "error",
      "no-eval": "error",
      "no-ex-assign": "error",
      "no-extend-native": "error",
      "no-extra-boolean-cast": "error",
      "no-fallthrough": "error",
      "no-func-assign": "error",
      "no-global-assign": "error",
      "no-import-assign": "error",
      "no-invalid-regexp": "error",
      "no-irregular-whitespace": "error",
      "no-iterator": "error",
      "no-labels": [
        "error",
        {
          "allowLoop": false,
          "allowSwitch": false,
        },
      ],
      "no-lone-blocks": "error",
      "no-loss-of-precision": "error",
      "no-multi-str": "error",
      "no-nested-ternary": "error",
      "no-new": "off",
      "no-new-func": "error",
      "no-new-native-nonconstructor": "error",
      "no-new-wrappers": "error",
      "no-obj-calls": "error",
      "no-proto": "error",
      "no-prototype-builtins": "error",
      "no-redeclare": [
        "error",
        {
          "builtinGlobals": false,
        },
      ],
      "no-regex-spaces": "error",
      "no-return-assign": [
        "error",
        "except-parens",
      ],
      "no-self-assign": [
        "error",
        {
          "props": true,
        },
      ],
      "no-self-compare": "error",
      "no-setter-return": "error",
      "no-shadow-restricted-names": "error",
      "no-sparse-arrays": "error",
      "no-template-curly-in-string": "error",
      "no-this-before-super": "error",
      "no-throw-literal": "error",
      "no-undef": "error",
      "no-unexpected-multiline": "error",
      "no-unreachable": "error",
      "no-unsafe-finally": "error",
      "no-unsafe-negation": "error",
      "no-unsafe-optional-chaining": "error",
      "no-unused-expressions": [
        "error",
        {
          "allowShortCircuit": true,
          "allowTaggedTemplates": true,
          "allowTernary": true,
        },
      ],
      "no-unused-vars": [
        "error",
        {
          "args": "after-used",
          "vars": "all",
        },
      ],
      "no-useless-call": "error",
      "no-useless-catch": "error",
      "no-useless-constructor": "error",
      "no-useless-escape": "error",
      "no-useless-rename": "error",
      "no-var": "warn",
      "no-void": "error",
      "no-with": "error",
      "prefer-exponentiation-operator": "error",
      "prefer-promise-reject-errors": "error",
      "prefer-rest-params": "error",
      "promise/param-names": "error",
      "require-yield": "error",
      "symbol-description": "error",
      "unicode-bom": "off",
      "use-isnan": [
        "error",
        {
          "enforceForIndexOf": true,
          "enforceForSwitchCase": true,
        },
      ],
      "valid-typeof": [
        "error",
        {
          "requireStringLiterals": true,
        },
      ],
      "yoda": [
        "error",
        "never",
      ],
    },
  },
  "warnings": [
    "unsupported rule: object-shorthand",
    "unsupported rule: accessor-pairs",
    "unsupported rule: array-bracket-spacing",
    "unsupported rule: arrow-spacing",
    "unsupported rule: block-spacing",
    "unsupported rule: brace-style",
    "unsupported rule: camelcase",
    "unsupported rule: comma-dangle",
    "unsupported rule: comma-spacing",
    "unsupported rule: comma-style",
    "unsupported rule: computed-property-spacing",
    "unsupported rule: curly",
    "unsupported rule: dot-location",
    "unsupported rule: dot-notation",
    "unsupported rule: eol-last",
    "unsupported rule: func-call-spacing",
    "unsupported rule: generator-star-spacing",
    "unsupported rule: indent",
    "unsupported rule: key-spacing",
    "unsupported rule: keyword-spacing",
    "unsupported rule: lines-between-class-members",
    "unsupported rule: multiline-ternary",
    "unsupported rule: new-parens",
    "unsupported rule: no-dupe-args",
    "unsupported rule: no-useless-backreference",
    "unsupported rule: no-extra-bind",
    "unsupported rule: no-extra-parens",
    "unsupported rule: no-floating-decimal",
    "unsupported rule: no-implied-eval",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: no-mixed-operators",
    "unsupported rule: no-mixed-spaces-and-tabs",
    "unsupported rule: no-multi-spaces",
    "unsupported rule: no-multiple-empty-lines",
    "unsupported rule: no-new-object",
    "unsupported rule: no-new-symbol",
    "unsupported rule: no-octal",
    "unsupported rule: no-octal-escape",
    "unsupported rule: no-sequences",
    "unsupported rule: no-tabs",
    "unsupported rule: no-trailing-spaces",
    "unsupported rule: no-undef-init",
    "unsupported rule: no-unmodified-loop-condition",
    "unsupported rule: no-unneeded-ternary",
    "unsupported rule: no-unreachable-loop",
    "unsupported rule: no-use-before-define",
    "unsupported rule: no-useless-computed-key",
    "unsupported rule: no-useless-return",
    "unsupported rule: no-whitespace-before-property",
    "unsupported rule: object-curly-newline",
    "unsupported rule: object-curly-spacing",
    "unsupported rule: object-property-newline",
    "unsupported rule: one-var",
    "unsupported rule: operator-linebreak",
    "unsupported rule: padded-blocks",
    "unsupported rule: prefer-const",
    "unsupported rule: prefer-regex-literals",
    "unsupported rule: quote-props",
    "unsupported rule: quotes",
    "unsupported rule: rest-spread-spacing",
    "unsupported rule: semi",
    "unsupported rule: semi-spacing",
    "unsupported rule: space-before-blocks",
    "unsupported rule: space-before-function-paren",
    "unsupported rule: space-in-parens",
    "unsupported rule: space-infix-ops",
    "unsupported rule: space-unary-ops",
    "unsupported rule: spaced-comment",
    "unsupported rule: template-curly-spacing",
    "unsupported rule: template-tag-spacing",
    "unsupported rule: wrap-iife",
    "unsupported rule: yield-star-spacing",
    "unsupported rule: import/no-absolute-path",
    "unsupported rule: n/handle-callback-err",
    "unsupported rule: n/no-callback-literal",
    "unsupported rule: n/no-deprecated-api",
    "unsupported rule: n/no-exports-assign",
    "unsupported rule: n/no-new-require",
    "unsupported rule: n/no-path-concat",
    "unsupported rule: n/process-exit-as-throw",
    "unsupported rule: block-scoped-var",
    "unsupported rule: func-name-matching",
    "unsupported rule: n/global-require",
    "unsupported rule: n/no-extraneous-require",
    "unsupported rule: n/no-missing-require",
    "unsupported rule: n/no-unpublished-require",
    "unsupported rule: n/no-unsupported-features/es-builtins",
    "unsupported rule: n/no-unsupported-features/es-syntax",
    "unsupported rule: n/no-unsupported-features/node-builtins",
    "unsupported rule: n/prefer-node-protocol",
    "unsupported rule: no-invalid-this",
    "unsupported rule: no-lonely-if",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: no-shadow",
    "unsupported rule: no-useless-return",
    "unsupported rule: object-shorthand",
    "unsupported rule: prefer-arrow-callback",
    "unsupported rule: prefer-let/prefer-let",
    "unsupported rule: promise/no-multiple-resolved",
  ],
}
`;
